src/utils/newcpp/algorithm.h

```diff
@@ -0,0 +1,113 @@
+#pragma once
+
+#include "cris/core/utils/newcpp/execution.h"
+
+#include <algorithm>
+#include <execution>
+#include <utility>
+
+namespace std {
+
+// NOLINTNEXTLINE(cppcoreguidelines-macro-usage,-warnings-as-errors)
+#define _CRIS_ALGORITHM_DEFINE_ADAPTER(ns, func)                                      \
```

Comment:
Put this outside of namespace.
Macro definition has nothing to do with scope, so don't leave a false impression of that.